src/MiddleMail/MiddleMailService.cs

```diff
@@ -14,17 +18,32 @@ namespace MiddleMail {
 	/// Implements a graceful shutdown by waiting for all processing tasks to finish work if cancellation is requested.
 	/// </summary>
 	public class MiddleMailService : BackgroundService {
+		public static readonly TimeSpan RateLimitWindow = TimeSpan.FromMinutes(1);
+		public static readonly TimeSpan RateLimitDelay = TimeSpan.FromMinutes(1);
 		
 		private IMessageProcessor processor;
 		private readonly ILogger<MiddleMailService> logger;
 		private readonly IMessageSource messageSource;
 		private int consumerTasksPending;
+		private readonly MiddleMailOptions options;
+
+		private readonly RateLimiter? rateLimiter;
 
-		public MiddleMailService(IMessageProcessor processor, ILogger<MiddleMailService> logger, IMessageSource messageSource) {
+		public MiddleMailService(IOptions<MiddleMailOptions> options, IMessageProcessor processor, ILogger<MiddleMailService> logger, IMessageSource messageSource) {
 			this.processor = processor;
 			this.logger = logger;
 			this.consumerTasksPending = 0;
 			this.messageSource = messageSource;
+			this.options = options.Value;
+
+			if (this.options.RateLimited) {
+				rateLimiter = new FixedWindowRateLimiter(
```

Comment:
I used fixed window largely because it seemed like the additional complexity of sliding window or token bucket wasn't worth it to worry about.
That said, I think changing to a sliding window makes plenty of sense, now that I look at it again. Would you prefer a sliding window?